src/ExtMongoDb.php

```diff
@@ -134,6 +134,8 @@ protected function internalGetItem(& $normalizedKey, & $success = null, & $casTo
             return;
         }
 
+        self::ensureArrayType($result);
```

Comment:
This is needed not only for the following `array_key_exists` call, but also for array subtypes which, without it, are kept `\MongoDB\Model\BSONDocument`